cmd/kubeapps-apis/server/packages.go

```diff
@@ -118,6 +161,125 @@ func (s packagesServer) GetAvailablePackageDetail(ctx context.Context, request *
 	}, nil
 }
 
+// GetInstalledPackageSummaries returns the installed package summaries based on the request.
+func (s packagesServer) GetInstalledPackageSummaries(ctx context.Context, request *packages.GetInstalledPackageSummariesRequest) (*packages.GetInstalledPackageSummariesResponse, error) {
+	contextMsg := ""
+	if request.Context != nil {
+		contextMsg = fmt.Sprintf("(cluster=[%s], namespace=[%s])", request.Context.Cluster, request.Context.Namespace)
+	}
+
+	log.Infof("+core GetInstalledPackageSummaries %s", contextMsg)
+
+	// Aggregate the response for each plugin
+	pkgs := []*packages.InstalledPackageSummary{}
+	// TODO: We can do these in parallel in separate go routines.
+	for _, p := range s.plugins {
+		response, err := p.server.GetInstalledPackageSummaries(ctx, request)
+		if err != nil {
+			return nil, status.Errorf(status.Convert(err).Code(), "Invalid GetInstalledPackageSummaries response from the plugin %v: %v", p.plugin.Name, err)
+		}
+
+		// Add the plugin for the pkgs
+		pluginPkgs := response.InstalledPackageSummaries
+		for _, r := range pluginPkgs {
+			if r.InstalledPackageRef == nil {
+				r.InstalledPackageRef = &packages.InstalledPackageReference{}
+			}
+			r.InstalledPackageRef.Plugin = p.plugin
+		}
+		pkgs = append(pkgs, pluginPkgs...)
+	}
+
+	From(pkgs).
+		// Order by package name, regardless of the plugin
+		OrderBy(func(pkg interface{}) interface{} {
+			return pkg.(*packages.InstalledPackageSummary).Name + pkg.(*packages.InstalledPackageSummary).InstalledPackageRef.Plugin.Name
+		}).
+		ToSlice(&pkgs)
```

Comment:
When we do the pagination correctly, we'll need different sortiing here anyway (since we'll need to keep pulling from each plugin and appending the next item in order. Sounds fun :)